src/strands/session/session_manager.py

```diff
@@ -71,3 +101,41 @@ def initialize(self, agent: "Agent", **kwargs: Any) -> None:
             agent: Agent to initialize
             **kwargs: Additional keyword arguments for future extensibility.
         """
+
+    def write_multi_agent_json(self, source: "MultiAgentBase") -> None:
+        """Write multi-agent state to persistent storage.
+
+        Args:
+            source: Multi-agent source object to persist
+        """
+        raise NotImplementedError(
+            f"{self.__class__.__name__} does not support multi-agent persistence "
+            "(write_multi_agent_json). Provide an implementation or use a "
+            "SessionManager with session_type=SessionType.MULTI_AGENT."
+        )
+
+    def read_multi_agent_json(self) -> dict[str, Any]:
```

Comment:
Update this signature to match initialize:

```
SessionManager.initialize_multi_agent(self, source: "MultiAgentBase", **kwargs: Any) -> None:)
```

Per the existing conventions, it shouldn't be returning the JSON but the implementation should handle reading the json itself. 

----

Returning a value would have made more sense if the existing methods did the same.  Let's discuss offline if this conflicts with other comments/guidance